src/cmd/services/m3comparator/main/querier.go

```diff
@@ -218,7 +250,44 @@ func (q *querier) generateRandomIters(
 		seriesList = append(seriesList, series)
 	}
 
-	return buildSeriesIterators(seriesList, query.Start, blockSize, opts)
+	return seriesList, nil
+}
+
+func (q *querier) generateMultiSeriesMetrics(
+	metricsName string,
+	start time.Time,
+	end time.Time,
+	resolution time.Duration,
+	blockSize time.Duration,
+) ([]series, error) {
+	suffix := strings.TrimPrefix(metricsName, "multi_")
+	seriesCount, err := strconv.Atoi(suffix)
+	if err != nil {
+		return nil, err
+	}
+
+	q.Lock()
+	defer q.Unlock()
+	rand.Seed(start.Unix())
+
+	seriesList := make([]series, seriesCount)
```

Comment:
nit: We typically favor `append` based slice building as it tends to be a little safer/more futureproof than index-based and [the performance difference is negligible](https://gist.github.com/xogeny/b819af6a0cf8ba1caaef)